### ISSUE TITLE
Provide support for non standard CSS properties and restrictions in VSCode

### ISSUE THREAD

**Issue by SinZ163** [Microsoft/vscode-css-languageservice#32]
Hi, I've been using VSCode a bunch for Dota 2 Custom Games, and it does the job great, except when it comes to how Valve have altered CSS to suit their needs.

A list of their properties can be found [here](https://developer.valvesoftware.com/wiki/Dota_2_Workshop_Tools/Panorama/CSS_Properties), and having the ability to enhance / replace the current CSS Schema with a new properties and restrictions, such as `hue-rotation` and hex based color inputs accepting 8 hexadecimal characters for color (`#333333b0`), as opposed to the normal 6 (`#FF0000`). 

Note I'm not asking to have Valves CSS properties being a part of VSCode, rather providing the infrastructure to allow users to enhance the existing language service with new properties or features.

**Comment by praveenpuglia** [comment]
I think this can be a plugin of some sort.

**Comment by aeschli** [comment]
Yes, the idea of Microsoft/vscode-css-languageservice is that it is to be resused for css like languages. The 8 digit hex colors might come soon for css as well, so I'm happy to take a PR that adds that.
I'm also open to make the properties configurable

**Comment by praveenpuglia** [comment]
I started working on a branch but I am not sure how do I test my changes in the editor. 

What I can't figure out is...
- How to get tell the editor that it's a valid color and don't show red underline.
- How to get the editor to show a box that shows the color beside the color definition.

https://github.com/praveenpuglia/vscode-css-languageservice/commit/0addc3d91f46eabaca19de22e3f63e08ec2ab622

**Comment by aeschli** [comment]
The fix in parser. _parseHexColor should fix the validation as well as making the color to reach the client. But on the css client side you'll have to turn the color into a valid css color that Chromium understands: https://github.com/Microsoft/vscode/blob/master/extensions/css/client/src/colorDecorators.ts#L114 

To run VSCode with your version of  checked out in the same folder
- have the vscode repo and the vscode-css-languageservice checked out in the same folder
- see https://github.com/Microsoft/vscode/wiki/How-to-Contribute for instructions how to selfhost vscode
- in the vscode workspace, cd to `extensions/css/server/` and run `npm run install-service-local` to install your local vscode-css-languageservice node modules

**Comment by praveenpuglia** [comment]
Alright! So I think I got a color box to appear beside a color but I am still failing validation.

I still get the validation error "Hex colors must have three or six hex numbers". Guess I have to change the validation here?

https://github.com/Microsoft/vscode-css-languageservice/blob/master/src/services/lint.ts#L530-L532

Changed the condition to 
```js
if (text.length !== 9 && text.length !== 7 && text.length !== 5 && text.length !== 4) {
    this.addEntry(node, lintRules_1.Rules.HexColorLength);
}
```
but still getting the validation error.
